### ISSUE TITLE
Error 500 when exporting route with empty POI name

### ISSUE THREAD

**Issue by Cuervino** [nrenner/brouter-web#427]
It lookis like it's not possible to export a file?

https://user-images.githubusercontent.com/28654802/120922638-a17dc600-c6ca-11eb-9f32-02141f1eda04.mp4

**Comment by bagage** [comment]
Hello, thanks for reporting this issue! Could you provide some additional info?
* route URL with the issue (might help)
* browser name and version
So that we can pinpoint the issue more precisely :).

**Comment by Cuervino** [comment]
Thanks for the reply!

I suppressed the 4 POIs, kept only the track, and it worked this time (so the bug priority is rather low).

I suspecting that the POIs were messing the export :-)

For info, here was the original url:
`https://brouter.de/brouter-web/#map=14/48.7247/2.1942/cyclosm,route-quality&lonlats=2.149368,48.804955;2.1636,48.776823;2.163706,48.774633;2.169027,48.773388;2.168964,48.768537;2.16849,48.767575;2.169263,48.767208;2.168604,48.765754;2.169006,48.766048;2.170282,48.765669;2.168352,48.764425;2.168926,48.763937;2.171742,48.762267;2.176511,48.759916;2.175282,48.758576;2.174456,48.758116;2.177566,48.758481;2.176918,48.757801;2.176131,48.757109;2.175744,48.754817;2.174218,48.75077;2.16581,48.738815;2.169734,48.737017;2.168494,48.733406;2.175058,48.729573;2.17444,48.729498;2.20012,48.721491;2.204516,48.722367;2.201234,48.722962;2.202221,48.73321;2.191436,48.738023;2.207739,48.740867;2.214343,48.740738;2.209372,48.744588;2.175046,48.762406;2.144801,48.770814;2.1321,48.777291;2.134674,48.780204;2.149282,48.801832&pois=2.133737,48.799102,Versailles;2.17559,48.72891,Point%20de%20d%C3%A9part;2.18891,48.74303,;2.18996,48.73903,Vauhallan;2.202652,48.728097,Abbaye%20de%20Limon;2.203714,48.721044,D36`

The url that worked:
`https://brouter.de/brouter-web/#map=14/48.7442/2.1674/cyclosm,route-quality&lonlats=2.149368,48.804955;2.1636,48.776823;2.163706,48.774633;2.169027,48.773388;2.168964,48.768537;2.16849,48.767575;2.169263,48.767208;2.168604,48.765754;2.169006,48.766048;2.170282,48.765669;2.168352,48.764425;2.168926,48.763937;2.171742,48.762267;2.176511,48.759916;2.175282,48.758576;2.174456,48.758116;2.177566,48.758481;2.176918,48.757801;2.176131,48.757109;2.175744,48.754817;2.174218,48.75077;2.16581,48.738815;2.169734,48.737017;2.168494,48.733406;2.175058,48.729573;2.17444,48.729498;2.20012,48.721491;2.204516,48.722367;2.201234,48.722962;2.202221,48.73321;2.191436,48.738023;2.207739,48.740867;2.214343,48.740738;2.209372,48.744588;2.175046,48.762406;2.144801,48.770814;2.1321,48.777291;2.134674,48.780204;2.149282,48.801832&pois=2.133737,48.799102,Versailles`

I really suspect the POIs with French accent to be the issue, but I could not really reproduce the issue.

My browser is FF89.0 (64-bit)

**Comment by bagage** [comment]
Oh indeed, thanks! You have a `POI` with no name, and it breaks the server. Oops.

**Comment by bagage** [comment]
Added a check on engine side as well as on front-end (though you can still remove it in URL). Should be better in the future, thanks :).

**Comment by Cuervino** [comment]
Thanks a lot, 
Have a nice day !

**Comment by Cuervino** [comment]
In fact, my scenario was a little bit more complex:
1. I uploaded a gpx file, that has a track and 5 POIs (and one was without name)
2. I made my own itinerary 
3. I unchecked the uploaded track on the side panel > it didn't remove the POIs, but only removed (visually) the uploaded track (I'm not 100% sure it's the expected behavior)
4. I exported my itinerary. The uploaded POIs were considered as being part of my itinerary (even if I did not care anymore about them), and the export bug happened.

So the check on the POI name emptyness should not only happen when the POI are manually created, but also when they are coming from an upload :-)
